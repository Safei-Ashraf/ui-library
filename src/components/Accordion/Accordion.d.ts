import { ReactNode } from "react";

interface AccordionItem {
	heading: string;
	content: string;
	id: string;
}

export interface AccordionProps {
	items: AccordionItem[];
	children?: ReactNode;
}
