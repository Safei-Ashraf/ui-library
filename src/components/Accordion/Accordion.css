.accordion-container {
	@apply list-none pt-xxs;
}
.accordion-item {
	@apply pr-md;
}
.accordion-heading {
	@apply w-full flex justify-between items-center  my-lg pb-md  text-primary text-fs.20 h-8 pr-md  
	border-solid border-0 border-b-thin border-neutral-lighter;
}
.accordion-content {
	@apply text-left text-fs.base text-neutral-focus;
}
