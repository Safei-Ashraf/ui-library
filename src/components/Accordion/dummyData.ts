export const items = [
	{
		id: "1",
		heading: "1st Heading here",
		content: `
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        lorem ipsum dolor sit amet lorem*10   Lorem ipsum dolor sit amet
        `,
	},
	{
		id: "2",
		heading: "2nd Heading here",
		content: "2nd content here 2nd content here",
	},
	{
		id: "3",
		heading: "3rd Heading here",
		content: `3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here
		3rd content here 3rd content here`,
	},
];
