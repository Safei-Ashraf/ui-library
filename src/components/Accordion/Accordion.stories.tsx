import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Accordion } from "./Accordion";

export default {
	title: "UI-Library/Accordion",
	component: Accordion,
	argTypes: {},
} as ComponentMeta<typeof Accordion>;

const Template: ComponentStory<typeof Accordion> = (args) => (
	<Accordion {...args} />
);

const items = [
	{
		id: "1",
		heading: "1st Heading here",
		content: "1st content here 1st content here",
	},
	{
		id: "2",
		heading: "2nd Heading here",
		content: "2nd content here 2nd content here",
	},
	{
		id: "3",
		heading: "3rd Heading here",
		content: "3rd content here 3rd content here",
	},
];
export const AccordionMain = Template.bind({});
AccordionMain.args = {
	items: items,
};
