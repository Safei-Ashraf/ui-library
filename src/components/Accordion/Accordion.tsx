import { useState } from "react";
import { AccordionProps } from "./Accordion.d";
import ChevronIcon from "./ChevronIcon";
import "./Accordion.css";

export const Accordion = ({ items, ...props }: AccordionProps): JSX.Element => {
	const [expandedIndex, setExpandedIndex] = useState(-1);
	const [rotateChevron, setRotateChevron] = useState(false);
	const handleRotate = () => setRotateChevron(!rotateChevron);

	const renderedItems = items.map(({ id, heading, content }, index) => {
		const isExapnded = index === expandedIndex;
		return (
			<li key={id}>
				<div
					className="accordion-heading"
					onClick={() => {
						if (isExapnded === true) {
							setExpandedIndex(-1);
						} else {
							setExpandedIndex(index);
						}
						handleRotate();
					}}
				>
					<h3 className="heading-text">{heading}</h3>
					<div className="expand-icon">
						<ChevronIcon
							direction="right"
							style={{
								transform: `${
									isExapnded
										? "rotate(0deg)"
										: "rotate(90deg)"
								}`,
								transition: "all 0.2s linear",
							}}
						/>
					</div>
				</div>
				{isExapnded && (
					<div className="accordion-content">{content}</div>
				)}
			</li>
		);
	});
	return (
		<ul {...props} className="accordion-container">
			{renderedItems}
		</ul>
	);
};
