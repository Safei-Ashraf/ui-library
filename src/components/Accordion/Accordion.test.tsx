import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import Accordion from "./index";

//Snapshot
describe("renders correctly", () => {
	const tree = renderer.create(<Accordion />).toJSON();
	expect(tree).toMatchSnapshot();
});

//Unit
describe("Accordion", () => {
	it("render Accordion", () => {
		render(<Accordion />);

		expect(screen.getByText("Accordion")).toBeInTheDocument();
	});
});
