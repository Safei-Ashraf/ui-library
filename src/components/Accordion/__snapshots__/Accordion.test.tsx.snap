// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ul
  className="accordion-container"
>
  <li
    className="accordion-item"
  >
    <div
      className="accordion-heading"
      onClick={[Function]}
    >
      <h3
        className="heading-text"
      >
        1st Heading here
      </h3>
      <div
        className="expand-icon"
      >
        <svg
          className="rotate-270"
          fill="currentColor"
          height="20"
          role="icon"
          style={
            Object {
              "transform": "rotate(90deg)",
              "transition": "all 0.1s linear",
            }
          }
          viewBox="0 0 455 455"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M227.5,0C101.855,0,0,101.855,0,227.5S101.855,455,227.5,455S455,353.145,455,227.5S353.145,0,227.5,0z M227.5,
 327.148L99.411,199.476l21.178-21.248L227.5,284.791l106.911-106.563l21.178,21.248L227.5,327.148z"
          />
        </svg>
      </div>
    </div>
  </li>
  <li
    className="accordion-item"
  >
    <div
      className="accordion-heading"
      onClick={[Function]}
    >
      <h3
        className="heading-text"
      >
        2nd Heading here
      </h3>
      <div
        className="expand-icon"
      >
        <svg
          className="rotate-270"
          fill="currentColor"
          height="20"
          role="icon"
          style={
            Object {
              "transform": "rotate(90deg)",
              "transition": "all 0.1s linear",
            }
          }
          viewBox="0 0 455 455"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M227.5,0C101.855,0,0,101.855,0,227.5S101.855,455,227.5,455S455,353.145,455,227.5S353.145,0,227.5,0z M227.5,
 327.148L99.411,199.476l21.178-21.248L227.5,284.791l106.911-106.563l21.178,21.248L227.5,327.148z"
          />
        </svg>
      </div>
    </div>
  </li>
  <li
    className="accordion-item"
  >
    <div
      className="accordion-heading"
      onClick={[Function]}
    >
      <h3
        className="heading-text"
      >
        3rd Heading here
      </h3>
      <div
        className="expand-icon"
      >
        <svg
          className="rotate-270"
          fill="currentColor"
          height="20"
          role="icon"
          style={
            Object {
              "transform": "rotate(90deg)",
              "transition": "all 0.1s linear",
            }
          }
          viewBox="0 0 455 455"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M227.5,0C101.855,0,0,101.855,0,227.5S101.855,455,227.5,455S455,353.145,455,227.5S353.145,0,227.5,0z M227.5,
 327.148L99.411,199.476l21.178-21.248L227.5,284.791l106.911-106.563l21.178,21.248L227.5,327.148z"
          />
        </svg>
      </div>
    </div>
  </li>
</ul>
`;
